Time PHEP (#1)
nabobalis/standards: pheps/phep-9999.md

1. Existing core packages will be audited to see how complex their time objects are.
2. Plans to be drawn up on how to tackle all code changes.
3. Funding to be found for this work to go ahead.

For any new packages that are being developed or created, they are recommended to use `astropy.time.Time` from the beginning.

# Motivation

One of the strategic goals of the PyHC Project is to "Coordinate development across projects to minimize duplication of effort".

However, all of the core PyHC packages were created not only before the existence of PyHC.
But also before many core libraries existed within the wider Scientiﬁc Python ecosystem (from `numpy` to `scipy` to `astropy` to `spiceypy` etc) meant that all core PyHC packages had to create their own implementations of time (and other components like coordinates) due to necessity.

Now that upstream ecosystem has grown at a rapid pace, the need for PyHC core package specific implementations has reduced.
This actually covers at least three areas (time, coordinate and units), however the focus for this PHEP is to start at time as that is considered the "simplest" case for standardization.

# Rationale

> One of the main focuses within PyHC should be around the goal of reducing the amount of code that each package has to carry around and maintain to do exactly the same.

Reviewer: ```suggestion
One of the main focuses within PyHC should be around the goal of reducing the amount of code that each package has to carry around and maintain to do implement the same functionality.
```

> This means that the limited time that maintainers or library authors have to focus on their package should be around adding more interesting topics like new features.

Reviewer: ```suggestion
This means that the limited time that maintainers or library authors have to focus on their package would be able to be used for things like adding new heliophysics specific features.
```

> This makes the usage of `astropy` very ideal as they are backed with a larger community as well as core team that maintains the library.

Reviewer: ```suggestion
This makes the usage of `astropy` ideal as they are backed by a larger community as well as core team that maintains the library, and are well funded.
```

It already is the foundation for many packages within astronomy and provides a strong overlap with existing objectives of many packages within PyHC.

> Current community objections are:

Author: See if what I added helps.

- This would require funding and that might be difficult based on the currently available sources.
- Potential lack of innovation if packages use the same framework.
- Education of a new time object for existing users of PyHC packages.

> 

Author: Yeah, I was stumped on this.
